catalog/imaging/imaging-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,269 @@
+import org.codice.ddf.catalog.async.data.api.internal.ProcessRequest
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource
+import org.codice.ddf.catalog.async.data.api.internal.ProcessUpdateItem
+import org.codice.ddf.catalog.async.data.impl.ProcessRequestImpl
+import org.codice.imaging.nitf.core.image.ImageSegment
+import org.codice.imaging.nitf.render.NitfRenderer
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Unroll
+class NitfPostProcessPluginSpec extends Specification {
+
+    private static final String GEO_NITF = "/i_3001a.ntf"
+
+    private static final long TEST_SIZE = 5L * 1024L * 1024L
+
+    private NitfPostProcessPlugin nitfPostProcessPlugin = null
+
+    private ProcessRequest<ProcessCreateItem> createProcessRequest
+
+    private ProcessRequest<ProcessUpdateItem> updateProcessRequest
+
+    private Metacard metacard
+
+    private Metacard metacard1
+
+    private ProcessResource processResource
+
+    private ProcessResource processResource1
+
+    private ProcessCreateItem processCreateItem
+
+    private ProcessCreateItem processCreateItem1
+
+    private ProcessUpdateItem processUpdateItem
+
+    def setup() {
+        this.nitfPostProcessPlugin = new NitfPostProcessPlugin()
+        this.createProcessRequest = Mock(ProcessRequestImpl)
+        this.updateProcessRequest = Mock(ProcessRequestImpl)
+        this.metacard = Mock(Metacard)
+        this.metacard1 = Mock(Metacard)
+        this.processResource = Mock(ProcessResource)
+        this.processResource1 = Mock(ProcessResource)
+        this.processCreateItem = Mock(ProcessCreateItem)
+        this.processCreateItem1 = Mock(ProcessCreateItem)
+        this.processUpdateItem = Mock(ProcessUpdateItem)
+        def (resourceUri, resourceInputStream) = getTestResource(GEO_NITF)
+        _ * metacard.getId() >> "0000000"
+        _ * metacard1.getId() >>"1111111"
+        _ * processResource.getName() >> "101ABC"
+        _ * processResource.getUri() >> resourceUri
+        _ * processResource.getInputStream() >> resourceInputStream
+        _ * processResource.getMimeType() >> NitfPostProcessPlugin.NITF_MIME_TYPE.toString()
+        _ * processResource.getSize() >> TEST_SIZE
+
+        _ * processResource1.getName() >> ("101ABC")
+
+        _ * processCreateItem.getMetacard() >> metacard
+        _ * processCreateItem.getProcessResource() >> processResource
+
+        _ * processCreateItem1.getMetacard() >> metacard1
+
+        _ * createProcessRequest.getProcessItems() >> [processCreateItem]
+        _ * updateProcessRequest.getProcessItems() >> [processCreateItem]
+    }
+
+    def "Building derived image filename from \"#ftitle\"" (String ftitle, String expectedFname) {
+        setup:
+            NitfPostProcessPlugin plugin = new NitfPostProcessPlugin();
+            def qualifier = "original";
+
+        when: "building a derived image filename"
+            def derivedFname = plugin.buildDerivedImageTitle(ftitle, qualifier, "jpg");
```

Comment:
✏️ Should buildDerivedImageTitle be static and marked `@VisibleForTesting`?

Reply:
Done